UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs

```diff
@@ -27,460 +21,247 @@ namespace MLAgents
-        public string PreviousActionPlaceholderName = "prev_action";
-
-        /// Name of the action mask node
-        private string ActionMaskPlaceholderName = "action_masks";
-        
-#if ENABLE_TENSORFLOW
-        TFGraph graph;
-        TFSession session;
-        bool hasRecurrent;
-        bool hasState;
-        bool hasBatchSize;
-        bool hasPrevAction;
-        bool hasMaskedActions; 
-        bool hasValueEstimate;
-        float[,] inputState;
-        int[,] inputPrevAction;
-        List<float[,,,]> observationMatrixList;
-        float[,] inputOldMemories;
-        float[,] maskedActions;
-        List<Texture2D> texturesHolder;
-        int memorySize;
-#endif
+        private bool _isControlled;
 
         /// <summary>
         /// When Called, the brain will be controlled externally. It will not use the
         /// model to decide on actions.
         /// </summary>
         public void SetToControlledExternally()
         {
-            isControlled = true;
+            _isControlled = true;
         }
         
+        /// <inheritdoc />
         protected override void Initialize()
         {
-#if ENABLE_TENSORFLOW
-#if UNITY_ANDROID && !UNITY_EDITOR
-// This needs to ba called only once and will raise an exception if 
-// there are multiple internal brains
-        try{
-            TensorFlowSharp.Android.NativeBinding.Init();
-        }
-        catch{
-            
-        }
-#endif
-
-            if (graphModel != null)
-            {
-                graph = new TFGraph();
-
-                graph.Import(graphModel.bytes);
-
-                session = new TFSession(graph);
-
-                if (graph[BatchSizePlaceholderName] != null)
-                {
-                    hasBatchSize = true;
-                }
-
-                if ((graph[RecurrentInPlaceholderName] != null) &&
-                    (graph[RecurrentOutPlaceholderName] != null))
-                {
-                    hasRecurrent = true;
-                    var runner = session.GetRunner();
-                    runner.Fetch(graph["memory_size"][0]);
-                    var networkOutput = runner.Run()[0].GetValue();
-                    memorySize = (int) networkOutput;
-                }
-
-                if (graph[VectorObservationPlacholderName] != null)
-                {
-                    hasState = true;
-                }
-
-                if (graph[PreviousActionPlaceholderName] != null)
-                {
-                    hasPrevAction = true;
-                }
-                if (graph["value_estimate"] != null)
-                {
-                    hasValueEstimate = true;
-                }
-                if (graph[ActionMaskPlaceholderName] != null)
-                {
-                    hasMaskedActions = true;
-                }
-            }
-
-
-            observationMatrixList = new List<float[,,,]>();
-            texturesHolder = new List<Texture2D>();
-#endif
+            SetModel(model);
```

Comment:
alluded to this earlier - calling setmodel using its own attribute is unintuitive.